tasks/deploy.js

```diff
@@ -0,0 +1,194 @@
+    console.table(finalizeVotingHelpers);
+
+    console.log("************************ Factories ************************");
+    console.table(finalizeFactories);
+
+    console.log("************************");
+    console.log(`DaoRegistry: ${dao.address}`);
+    console.log(`BankExtension: ${extensions.bank.address}`);
+    console.log(
+      `NFTExtension: ${extensions.nft ? extensions.nft.address : ""}`
+    );
+    console.log(
+      `ERC20Extension: ${
+        extensions.erc20Ext ? extensions.erc20Ext.address : ""
+      }`
+    );
+    console.log("************************");
+  } else {
+    console.log("************************");
+    console.log("no migration for network " + network);
+    console.log("************************");
+  }
+});
+
+const capitalize = str => str.charAt(0).toUpperCase() + str.slice(1);
+
+const formattedOject = (object, name) => {
+  const array = [];
+  for (let key in object) {
+    array.push({
+      name: capitalize(key) + name,
+      address: object[key] ? object[key].address : ""
+    });
+  }
+  return array;
+}
+
+async function deployTestDao(deployFunction, network, accounts, hardhatImports) {
+  if (!process.env.DAO_NAME)
+    throw Error("Missing env var: DAO_NAME");
+  if (!process.env.ERC20_TOKEN_NAME)
+    throw Error("Missing env var: ERC20_TOKEN_NAME");
+  if (!process.env.ERC20_TOKEN_SYMBOL)
+    throw Error("Missing env var: ERC20_TOKEN_SYMBOL");
+  if (!process.env.ERC20_TOKEN_DECIMALS)
+    throw Error("Missing env var: ERC20_TOKEN_DECIMALS");
+
+  return await deployDao({
+    ...hardhatImports,
+    deployFunction,
+    unitPrice: unitPrice,
+    nbUnits: numberOfUnits,
+    tokenAddr: ETH_TOKEN,
+    erc20TokenName: process.env.ERC20_TOKEN_NAME,
+    erc20TokenSymbol: process.env.ERC20_TOKEN_SYMBOL,
+    erc20TokenDecimals: process.env.ERC20_TOKEN_DECIMALS,
+    maxChunks: maximumChunks,
+    votingPeriod: 10, // 10 secs
+    gracePeriod: 1, // 1 sec
+    offchainVoting: true,
+    chainId: getNetworkDetails(network).chainId,
+    deployTestTokens: false,
+    finalize: false,
+    maxExternalTokens: 100,
+    couponCreatorAddress: "0x7D8cad0bbD68deb352C33e80fccd4D8e88b4aBb8",
+    offchainAdmin: "0xedC10CFA90A135C41538325DD57FDB4c7b88faf7",
+    daoName: process.env.DAO_NAME,
+    owner: accounts[0],
+  });
+}
+
+async function deployTestNetworkDao(deployFunction, network, hardhatImports) {
+  if (!process.env.DAO_NAME)
+    throw Error("Missing env var: DAO_NAME");
+  if (!process.env.DAO_OWNER_ADDR)
+    throw Error("Missing env var: DAO_OWNER_ADDR");
+  if (!process.env.ERC20_TOKEN_NAME)
+    throw Error("Missing env var: ERC20_TOKEN_NAME");
+  if (!process.env.ERC20_TOKEN_SYMBOL)
+    throw Error("Missing env var: ERC20_TOKEN_SYMBOL");
+  if (!process.env.ERC20_TOKEN_DECIMALS)
+    throw Error("Missing env var: ERC20_TOKEN_DECIMALS");
+  if (!process.env.COUPON_CREATOR_ADDR)
+    throw Error("Missing env var: COUPON_CREATOR_ADDR");
```

Comment:
Codacy has a fix for the issue: [Expected { after 'if' condition.](https://app.codacy.com/gh/openlawteam/tribute-contracts/pullRequest?prid=8819147)
```suggestion
    {throw Error("Missing env var: COUPON_CREATOR_ADDR");}
```